src/software/SfM/main_openMVG2ABC.cpp

```diff
@@ -0,0 +1,273 @@
+    for(const auto it: sfm_data.GetViews()) {
+        const View * view = it.second.get();
+        if (!sfm_data.IsPoseAndIntrinsicDefined(view))
+          continue;
+
+        // OpenMVG Camera
+        const openMVG::geometry::Pose3 pose = sfm_data.GetPoseOrDie(view);
+        auto iterIntrinsic = sfm_data.GetIntrinsics().find(view->id_intrinsic);
+        openMVG::cameras::Pinhole_Intrinsic *cam = static_cast<openMVG::cameras::Pinhole_Intrinsic*>(iterIntrinsic->second.get());
+        openMVG::Mat34 P = cam->get_projective_equivalent(pose);
+
+        // Extract internal matrix, rotation and translation
+        const openMVG::Mat3 R = pose.rotation();
+        const openMVG::Vec3 center = pose.center();
+
+        // POSE
+        // Compensate translation with rotation
+        // Build transform matrix
+        Abc::M44d xformMatrix;
+        xformMatrix[0][0]= R(0,0);
+        xformMatrix[0][1]= R(0,1);
+        xformMatrix[0][2]= R(0,2);
+        xformMatrix[1][0]= R(1,0);
+        xformMatrix[1][1]= R(1,1);
+        xformMatrix[1][2]= R(1,2);
+        xformMatrix[2][0]= R(2,0);
+        xformMatrix[2][1]= R(2,1);
+        xformMatrix[2][2]= R(2,2);
+        xformMatrix[3][0]= center(0);
+        xformMatrix[3][1]= center(1);
+        xformMatrix[3][2]= center(2);
+        xformMatrix[3][3]= 1.0;
+
+        // Correct camera orientation for alembic
+        M44d scale;
+        scale[1][1] = -1;
+        scale[2][2] = -1;
+        
+        xformMatrix = scale*xformMatrix;
+
+        XformSample xformsample;
+        xformsample.setMatrix(xformMatrix);
+
+        stringstream ss;
+        ss << stlplus::basename_part(view->s_Img_path); 
+        Alembic::AbcGeom::OXform xform(topObj, "camxform_" + ss.str());
+        xform.getSchema().set(xformsample);
+
+        // Camera intrinsic parameters
+        OCamera camObj(xform, "camera_" + ss.str());
+        CameraSample camSample;
+    
+        // Take the max of the image size to handle the case where the image is in portrait mode 
+        const float imgWidth = cam->w();  
+        const float imgHeight = cam->h(); 
+        const float sensorWidthPix = std::max(imgWidth, imgHeight);
+        const float sensorHeightPix = std::min(imgWidth, imgHeight);
+        const float focalLengthPix = cam->focal();
+        const float dx = cam->principal_point()(0);
+        const float dy = cam->principal_point()(1);
+        // Use a common sensor width as we don't have this information at this point
+        // We chose a full frame 24x36 camera
+        const float sensorWidth = 36.0; // 36mm per default
+        const float sensorHeight = sensorWidth*sensorHeightPix/sensorWidthPix;
+        const float focalLength = sensorWidth*focalLengthPix/sensorWidthPix;
+        // Following values are in cm, hence the 0.1 multiplier
+        const float hoffset = 0.1*sensorWidth*(0.5-dx/imgWidth);
+        const float voffset = 0.1*sensorHeight*(dy/imgHeight-0.5)*sensorHeightPix/sensorWidthPix;
```

Comment:
Why do you multiply by `sensorHeightPix/sensorWidthPix` here?
```
const float voffset = 0.1*sensorHeight*(dy/imgHeight-0.5);
```
Upper there is `sensorHeight = sensorWidth*sensorHeightPix/sensorWidthPix`. So it's like the ratio is applied twice.